src/server/pachyderm_test.go

```diff
@@ -882,13 +882,23 @@ func TestRunPipeline(t *testing.T) {
 		require.NoError(t, backoff.Retry(func() error {
 			return c.RunPipeline(pipeline, []*pfs.CommitProvenance{
 				client.NewCommitProvenance(dataRepo, branchA, commitA.ID),
-			})
+			}, "")
 		}, backoff.NewTestingBackOff()))
 
 		// the downstream pipeline shouldn't have any new jobs, since runpipeline jobs don't propagate
 		jobInfos, err = c.FlushJobAll([]*pfs.Commit{commitA}, []string{downstreamPipeline})
 		require.NoError(t, err)
 		require.Equal(t, 1, len(jobInfos))
+
+		// now rerun the one job that we saw
+		require.NoError(t, backoff.Retry(func() error {
+			return c.RunPipeline(downstreamPipeline, nil, jobInfos[0].Job.ID)
+		}, backoff.NewTestingBackOff()))
```

Comment:
Is there a reason this needs to be in a `backoff.Retry`? It seems like you should be able to just call it, since you know the `FlushJobAll` waits for the job to complete. I could be missing something though.